smarts/core/sensors.py

```diff
@@ -1300,8 +1304,18 @@ def __call__(
                 if lane.offset_along_lane(pt) >= lane_pos.s:
                     upcoming_signals.append(feat)
                     break
+        if self._include_foes:
+            self._find_foe_signals(lane, used_features, upcoming_signals)
+
         lookahead = self._lookahead - lane.length + lane_pos.s
-        self._find_signals_ahead(lane, lookahead, plan.route, upcoming_signals)
+        self._find_signals_ahead(
+            lane,
+            lookahead,
+            plan.route,
+            self._include_foes,
+            used_features,
+            upcoming_signals,
+        )
```

Comment:
It would be better to have functions with return values, compared to relying on functions with side effects.